modules/Slider.js

```diff
@@ -0,0 +1,217 @@
+      height:R.h-10,
+      steps:30,
+
+      dragableSlider:true,
+      dragRect:R,
+      useIncr:true,
+      useMap:false,
+      oversizeR:0,
+      oversizeL:0,
+      propagateDrag:false,
+      timeout:1,
+
+      drawableSlider:true,
+      colorFG:g.theme.fg2,
+      colorBG:g.theme.bg2,
+      rounded:true,
+      outerBorderSize:2,
+      innerBorderSize:2,
+
+      autoProgress:false,
+    },conf);
+
+    // If horizontal, flip things around.
+    if (o.c.horizontal) {
+      let mediator = o.c.xStart;
+      o.c.xStart = o.c.yStart;
+      o.c.yStart = mediator;
+      mediator = o.c.width;
+      o.c.width = o.c.height;
+      o.c.height = mediator;
+      delete mediator;
+    }
+
+    // Make room for the border. Underscore indicates the area for the actual indicator bar without borders.
+    let totalBorderSize = o.c.outerBorderSize + o.c.innerBorderSize;
+    o.c._xStart = o.c.xStart + totalBorderSize;
+    o.c._width = o.c.width - 2*totalBorderSize;
+    o.c._yStart = o.c.yStart + totalBorderSize;
+    o.c._height = o.c.height - 2*totalBorderSize;
+    o.c.rounded = o.c.rounded?20:0;
+
+    o.c.STEP_SIZE = ((!o.c.horizontal?o.c._height:o.c._width)-(!o.c.rounded?0:(2*o.c.rounded-7)))/o.c.steps;
+
+    // Add a rectangle object with x, y, x2, y2, w and h values.
+    o.c.r = {x:o.c.xStart, y:o.c.yStart, x2:o.c.xStart+o.c.width, y2:o.c.yStart+o.c.height, w:o.c.width, h:o.c.height};
+
+    // Initialize the level
+    o.v.level = (o.c.currLevel||o.c.currLevel===0)?o.c.currLevel:o.c.steps/2;
+
+    // Only add interactivity if wanted.
+    if (o.c.dragableSlider) { 
+      const Y_MAX = g.getHeight()-1; // Should this take users screen calibration into account?
+
+      o.v.ebLast = 0;
+      o.v.dy = 0;
+
+      o.f.wasOnDragRect = (exFirst, eyFirst)=>{
+        "ram";
+        return exFirst>o.c.dragRect.x && exFirst<o.c.dragRect.x2 && eyFirst>o.c.dragRect.y && eyFirst<o.c.dragRect.y2;
+      };
+
+      o.f.wasOnIndicator = (exFirst)=>{
+        "ram";
+        if (!o.c.horizontal) return exFirst>o.c._xStart-o.c.oversizeL*o.c._width && exFirst<o.c._xStart+o.c._width+o.c.oversizeR*o.c._width;
+        if (o.c.horizontal) return exFirst>o.c._yStart-o.c.oversizeL*o.c._height && exFirst<o.c._yStart+o.c._height+o.c.oversizeR*o.c._height;
+      };
+
+      // Function to pass to `Bangle.on('drag', )`
+      o.f.dragSlider = e=>{
+        "ram";
+        if (o.v.ebLast==0) {
+          exFirst = o.c.horizontal?e.y:e.x;
+          eyFirst = o.c.horizontal?e.x:e.y;
+        }
+
+        // Only react if on allowed area.
+        if (o.f.wasOnDragRect(exFirst, eyFirst)) {
+          o.v.dragActive = true;
+          if (!o.c.propagateDrag) E.stopEventPropagation&&E.stopEventPropagation();
+
+          if (o.v.timeoutID) {clearTimeout(o.v.timeoutID); o.v.timeoutID = undefined;}
+          if (e.b==0 && !o.v.timeoutID && (o.c.timeout || o.c.timeout===0)) o.v.timeoutID = setTimeout(o.f.remove, 1000*o.c.timeout);
+
+          o.v.cbObj;
```

Comment:
Sorry, one very minor thing! I think we can drop this line now :)

```suggestion
```

Reply:
Ah - of course! 🙃